room/room-compiler-processing/src/main/java/androidx/room/compiler/processing/XTypeElement.kt

```diff
@@ -157,4 +157,9 @@ interface XTypeElement : XHasModifiers, XElement, XMemberContainer {
      * List of interfaces implemented by this class
      */
     fun getSuperInterfaceElements(): List<XTypeElement>
+
+    /**
+     * Returns the list of all classes nested directly within this class.
+     */
+    fun getEnclosedTypeElements(): List<XTypeElement>
```

Comment:
will this include companions?
I'm fine either way but we should document it here and also add a test.

for the most part, we don't include them (e.g. static methods inherited from companion are just reported as static methods on the containing class)

Reply:
@yigit Good question. This does include companions; I've added a test and documentation.